src/domain/fx/cyril.js

```diff
@@ -250,16 +250,35 @@ const _getPositionChanges = async (commitRequestIdList, transferIdList, original
   for (const transferId of transferIdList) {
     const transferRecord = await TransferFacade.getById(transferId)
     const transferPositionChanges = await ParticipantPositionChangesModel.getReservedPositionChangesByTransferId(transferId)
-    transferPositionChanges.forEach((transferPositionChange) => {
+
+    // Context: processing interscheme transfer abort with accompanying fx transfer where the payee DFSP is proxied
+    //
+    // If the transferPositionChanges is empty and there is a commitRequestId and the tranferId is the same as the originalId,
+    // then it is a case where the transfer has no position change for the transfer in the buffer scheme but has position change for the fx transfer.
+    // In that case we need to add a zero-value position change for the transfer so that we can notify the payer and update the transfer state.
+    // This zero-value position change is only used for notification and transfer state change.
+    // It is not used for actual position change as it is excluded based on zero value here src/domain/position/abort.js:L81.
+    if (isAbort && transferRecord && transferRecord.payeeIsProxy && transferPositionChanges.length === 0 && !!commitRequestIdList.length && originalId === transferId) {
       positionChanges.push({
         isFxTransferStateChange: false,
         transferId,
         isOriginalId: originalId === transferId,
-        notifyTo: transferRecord.externalPayerName || transferRecord.payerFsp,
-        participantCurrencyId: transferPositionChange.participantCurrencyId,
-        amount: -transferPositionChange.change
+        notifyTo: transferRecord.payerFsp,
+        participantCurrencyId: transferRecord.payerParticipantCurrencyId,
```

Comment:
Does it need to be null?

Reply:
And is there any reason, you want to use the existing positionChanges array? I think its better to use a new field in context like `transferStateChanges` for this. Because positionChanges is used in the position handler to recursively change the positions by producing followup position messages. I am not sure if its a good idea to messup with this variable.